kafka/retry_lazy_client.go

```diff
@@ -0,0 +1,206 @@
+package kafka
+
+import (
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+type RetryLazyClient struct {
```

Comment:
there's a lot of boilerplate duplication in this file. Any way that can be reduced?